src/v/resource_mgmt/memory_groups.cc

```diff
@@ -45,9 +45,18 @@ struct memory_shares {
 
 } // namespace
 
+size_t
+compaction_memory_reservation::reserved_bytes(size_t total_memory) const {
+    return std::min(total_memory, max_bytes);
+}
+
 system_memory_groups::system_memory_groups(
-  size_t total_system_memory, bool wasm_enabled)
-  : _total_system_memory(total_system_memory)
+  size_t total_available_memory,
+  compaction_memory_reservation compaction,
+  bool wasm_enabled)
+  : _compaction_reserved_memory(
+    compaction.reserved_bytes(total_available_memory))
```

Comment:
i'm probably reading this incorrectly. but it looks like the compaction reserved memory is the minimum between the configured size for the index (e.g. 128mb) and the total available memory size on the shard? if so, then if a shard had less than 128mb of memory we'd allocate 100% of the available memory for compaction?

EDIT: mostly nevermind. I see in the next commit what's happening with the % config option.

Reply:
Yeah, this is just a (probably not very effective) safeguard against specifying a bogus high compaction memory config